src-electron/generator/helper-session.js

```diff
@@ -126,6 +126,29 @@ async function user_device_types(options) {
   return templateUtil.templatePromise(this.global, promise)
 }
 
+/**
+ * Creates iterator over endpoint composition requirements for a device type.
+ * This works inside user_device_types context where device type ref is available.
+ * Returns required device types that must be on separate endpoints.
+ * @param {*} options
+ */
+async function user_endpoint_composition_requirements(options) {
+  if (!this.deviceTypeRef) {
+    throw new Error(
+      'user_endpoint_composition_requirements must be called within user_device_types context'
+    )
+  }
+  let promise = queryDeviceType
+    .selectEndpointCompositionRequirementsByDeviceTypeRef(
+      this.global.db,
+      this.deviceTypeRef
+    )
+    .then((requirements) =>
```

Comment:
Have we not weeded out the ".then()" syntax in favor of the "await/async"?

I've been a bit disconnected from zap, but I remember that I spent a lot of time killing the more convoluted .then().then() syntax in favor of much more understandable "await/async" syntax?

Unless the coding standards in zap project have changed, we should NOT be adding more .then() syntax anywhere, and if anything we should be removing the one we find as regular refactoring.

Reply:
all the helpers are written with .then , we should go through them all and fix them in a separate PR